core/blockchain/impl/cached_tree.cpp

```diff
@@ -162,15 +179,49 @@ namespace kagome::blockchain {
   }
 
   bool TreeMeta::chooseBest(std::shared_ptr<TreeNode> node) {
-    auto best = best_leaf.lock();
+    if (node->reverted) {
+      return false;
+    }
+    auto best = best_block.lock();
     BOOST_ASSERT(best);
+    BOOST_ASSERT(not best->reverted);
     if (getWeight(node) > getWeight(best)) {
-      best_leaf = node;
+      best_block = node;
       return true;
     }
     return false;
   }
 
+  void TreeMeta::forceRefreshBest() {
+    auto root = last_finalized.lock();
+    std::multiset<std::shared_ptr<TreeNode>> candidates{};
+    for (auto &leaf : leaves) {
+      if (auto node = root->findByHash(leaf)) {
+        candidates.emplace(std::move(node));
+      }
+    }
+
+    auto best = std::move(root);
+    while (not candidates.empty()) {
+      auto node = candidates.extract((++candidates.rbegin()).base());
```

Comment:
```suggestion
      auto node = candidates.extract(std::prev(candidates.end()));
```

Reply:
https://en.cppreference.com/w/cpp/iterator/prev See **Notes**:

Although the expression --c.end() often compiles, it is not guaranteed to do so: c.end() is an rvalue expression, and there is no iterator requirement that specifies that decrement of an rvalue is guaranteed to work. In particular, when iterators are implemented as pointers or its operator-- is lvalue-ref-qualified, --c.end() does not compile, while std::prev(c.end()) does.